### PR TITLE
adding await to save button

### PR DESCRIPTION
**Pull request by UrsaStutsman**
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/summary/WorkspaceSummaryPage.scala [hunk 1]
```diff
@@ -63,6 +63,7 @@ class WorkspaceSummaryPage(namespace: String, name: String)(implicit webDriver:
 
     def clickEdit(): Unit = {
       editButton.doClick()
+      saveButton.awaitEnabledState()
```
Comment (breilly2) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/summary/WorkspaceSummaryPage.scala]: Ah, nice find!

So, this is an interesting page in that it switches between view/edit modes. The `awaitReady` function is not aware of these modes so it cannot behave appropriately.  I agree that waiting for the save button is a good solution.

My only question is whether it should be here or if `clickEdit` should manipulate a property of the page that `awaitReady` can use to decide what it's going to check. Advantage is that it cohesively describes the stateful nature of the page. Disadvantage is that is an extra level of indirection when tracing the linear behavior of a scenario.

I can't argue strongly for either case over the other at the moment. Curious to hear what others think. For now, either way is fine with me.

**Review by breilly2 — APPROVED** [review]
👍 See comment for a possible refactoring.

**Comment by aweng98** [comment]
sorry, I don't know which test this fix is for? Is this in jenkin log?

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/summary/WorkspaceSummaryPage.scala (hunk 1 above)
Comment (MattPutnam) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/summary/WorkspaceSummaryPage.scala]: I'm working on a better way to do this whole page. I think this is the most correct option with the way the code is right now.

**Review by aweng98 — APPROVED** [review]
LGTM.

**Pull request closed without merging**